hexrd/ui/mask_manager_dialog.py

```diff
@@ -310,12 +313,53 @@ def rebuild_masks(self):
         self.update_masks_list('raw')
 
     def masks_to_panel_buffer(self):
+        show_dialog = False
+        selection = 'Replace buffer'
+        for det in HexrdConfig().detectors.values():
+            buff_val = det.get('buffer', {}).get('value', None)
+            if isinstance(buff_val, np.ndarray) and buff_val.ndim == 2:
+                show_dialog = True
+                break
+
+        if show_dialog:
+            dialog = QDialog(self.ui)
+            layout = QVBoxLayout()
+            dialog.setLayout(layout)
+
+            options = QComboBox(dialog)
+            options.addItem('Replace buffer')
+            options.addItem('Logical AND with buffer')
+            options.addItem('Logical OR with buffer')
+            layout.addWidget(options)
+
+            buttons = QDialogButtonBox.Ok | QDialogButtonBox.Cancel
+            button_box = QDialogButtonBox(buttons, dialog)
+            button_box.accepted.connect(dialog.accept)
+            button_box.rejected.connect(dialog.reject)
+            layout.addWidget(button_box)
+
+            if not dialog.exec_():
+                # canceled
+                return
```

Comment:
FYI: I think you can pretty much do the same thing here with just [`QInputDialog.getItem()`](https://doc.qt.io/qt-5/qinputdialog.html#getItem). But you don't need to change this if it works.